com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceOperationsExecution.java

```diff
@@ -165,19 +176,14 @@ private ServiceOperation getLastDeleteServiceOperation(ExecutionWrapper executio
     private boolean isServiceDeleted(ExecutionWrapper execution, UUID uuid) {
         List<CloudEvent> serviceEvents = eventsGetter.getEvents(uuid, execution.getControllerClient());
         if (serviceEvents != null) {
-            return serviceEvents.stream()
-                .filter(e -> e.getTimestamp() != null)
-                .sorted((a, b) -> b.getTimestamp()
-                    .compareTo(a.getTimestamp()))
-                .findFirst()
-                .filter(e -> e.getType()
-                    .equals(SERVICE_EVENT_TYPE_DELETE))
-                .isPresent();
+            return serviceEvents.get(serviceEvents.size() - 1)
```

Comment:
Not like:
1. serviceEvents.get(serviceEvents.size() - 1) could return null -> NPE => SERVICE_EVENT_TYPE_DELETE.equals(serviceEvents.get(serviceEvents.size() - 1));
2. Order by ... DESC returned from the CC ?